pkg/resources/database_acceptance_test.go

```diff
@@ -1160,233 +1161,232 @@ resource "snowflake_database" "test" {
 				),
 			},
+			// Change in parameter shouldn't change the state Snowflake
 			{
-				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
-				Config:                   databaseStateUpgraderFromReplicaNewAfterUpgrade(id),
-				ExpectError:              regexp.MustCompile("failed to upgrade the state with database created from replica, please use snowflake_secondary_database or deprecated snowflake_database_old instead"),
-			},
-			{
-				ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
-				Config:                   databaseStateUpgraderFromReplicaNew(id, primaryDatabaseId),
-				ResourceName:             "snowflake_secondary_database.test",
-				ImportStateId:            id.FullyQualifiedName(),
-				ImportState:              true,
+				Config: databaseWithPublicSchemaConfig(id, false),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_database.test", "id", id.Name()),
+					containsPublicSchema(t, id),
+				),
 			},
 		},
 	})
 }
 
-func databaseStateUpgraderFromReplicaOld(id sdk.AccountObjectIdentifier, primaryDatabaseId sdk.ExternalObjectIdentifier) string {
-	return fmt.Sprintf(`
-resource "snowflake_database" "test" {
-	name = "%s"
-	data_retention_time_in_days = 0 # to avoid in-place update to -1
-	from_replica = %s
+func doesNotContainPublicSchema(t *testing.T, id sdk.AccountObjectIdentifier) resource.TestCheckFunc {
+	t.Helper()
+	return func(state *terraform.State) error {
+		if slices.ContainsFunc(acc.TestClient().Database.Describe(t, id).Rows, func(row sdk.DatabaseDetailsRow) bool { return row.Name == "PUBLIC" && row.Kind == "SCHEMA" }) {
+			return fmt.Errorf("expected database %s to not contain public schema", id.FullyQualifiedName())
+		}
+		return nil
+	}
 }
-`, id.Name(), strconv.Quote(primaryDatabaseId.FullyQualifiedName()))
+
+func containsPublicSchema(t *testing.T, id sdk.AccountObjectIdentifier) resource.TestCheckFunc {
+	t.Helper()
+	return func(state *terraform.State) error {
+		if !slices.ContainsFunc(acc.TestClient().Database.Describe(t, id).Rows, func(row sdk.DatabaseDetailsRow) bool { return row.Name == "PUBLIC" && row.Kind == "SCHEMA" }) {
+			return fmt.Errorf("expected database %s to contain public schema", id.FullyQualifiedName())
+		}
+		return nil
+	}
 }
 
-func databaseStateUpgraderFromReplicaNewAfterUpgrade(id sdk.AccountObjectIdentifier) string {
+func databaseWithDropPublicSchemaConfig(id sdk.AccountObjectIdentifier, withDropPublicSchema bool) string {
 	return fmt.Sprintf(`
 resource "snowflake_database" "test" {
 	name = "%s"
-	data_retention_time_in_days = 0
+	drop_public_schema_on_creation = %s
 }
-`, id.Name())
+`, id.Name(), strconv.FormatBool(!withPublicSchema))
 }
 
-func databaseStateUpgraderFromReplicaNew(id sdk.AccountObjectIdentifier, primaryDatabaseId sdk.ExternalObjectIdentifier) string {
-	return fmt.Sprintf(`
-resource "snowflake_secondary_database" "test" {
-	name = "%s"
-	as_replica_of = %s
+/*
```

Comment:
Thanks for addding the manual steps that currently can't be run using this testing framework.

I think that we should no have the directly as comment in the acceptance test but maybe rather let's add manual tests directory (or something similar), let's add a short readme why we do this, and there let's add them as separate files.

WDYT @sfc-gh-jcieslak @sfc-gh-jmichalak ?

Reply:
Good idea, maybe some of them could be automatic after upgrading to plugin framework.

Reply:
Added